plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/vault/VaultAuthenticatorWithExpiration.java

```diff
@@ -0,0 +1,45 @@
+package io.jenkins.plugins.casc.impl.secrets.vault;
+
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultException;
+
+import java.util.Calendar;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+abstract class VaultAuthenticatorWithExpiration implements VaultAuthenticator {
+
+    private final static Logger LOGGER = Logger.getLogger(VaultAuthenticatorWithExpiration.class.getName());
+
+    private Calendar tokenExpiration;
+
+    public boolean isTokenTTLExpired() {
+        if (tokenExpiration == null) return true;
+
+        boolean result = true;
+        Calendar now = Calendar.getInstance();
+        long timeDiffInMillis = now.getTimeInMillis() - tokenExpiration.getTimeInMillis();
+        if (timeDiffInMillis < -2000L) {
+            // token will be valid for at least another 2s
+            result = false;
+            LOGGER.log(Level.FINE, "Auth token is still valid");
+        } else {
+            LOGGER.log(Level.FINE, "Auth token has to be re-issued" + timeDiffInMillis);
+        }
+
+        return result;
+    }
+
+    public void getTTLExpiryOfCurrentToken(Vault vault) {
+        try {
+            // save token TTL
+            tokenExpiration = Calendar.getInstance();
+            tokenExpiration.add(Calendar.SECOND, (int) vault.auth().lookupSelf().getTTL());
+        } catch (VaultException e) {
+            LOGGER.log(Level.WARNING, "Could not determine token expiration. " +
+                    "Check if token is allowed to access auth/token/lookup-self. " +
+                    "Assuming token TTL expired.", e);
+            tokenExpiration = Calendar.getInstance();
```

Comment:
Get rid of the potential of calling `Calendar.getInstance();` twice because of an `VaultException`
```suggestion
```

Reply:
Fixed